### PR TITLE
(#429) CMS Translate Infographic tests 

### LINKED ISSUE #429: CMS: Translate Infographic
## As a [role], I want [feature], so that [benefit]

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario: Create a Spanish Infographic translation
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on "Media" sub tab
        Then user selects "Translate" option from Operations dropdown for media with title "Automated Test Infographic"
        Then the page title is "Translations of Automated Test Infographic"
        When user clicks on "Add" button to add translation
        Then page title is "Crear traducción Español de Automated Test Infographic"
        And user clears out "Pretty URL" field
        And user clears out "Nombre" field
        And user clears out "Card Title" field
        And user clears out "Feature Card Description" field
        And user fills out the following fields
            | fieldLabel               | value                                                   | field_name                     |
            | Pretty URL               | test-infographic-spanish                                | field_pretty_url               |
            | Nombre                   | Automated Test Infographic Spanish                      | name[0][value]                 |
            | Card Title               | Automated Test Infographic Spanish - Card Title         | field_card_title               |
            | Feature Card Description | Automated Test Infographic Spanish - Featured Card Desc | field_feature_card_description |
        And user clears out "List Description" field
        And user enters "Automated Test Infographic List Description Spanish" into "List Description" text field
        And user selects "Review" from "Change to" dropdown
        Then user saves the content page
        And user clicks on "Moderated media" sub tab
        Then user selects "Editar" option from Operations dropdown for media with title "Test Video_Spanish"
        Then page title is "Automated Test Infographic_Spanish [Español traducción]"
        And user selects "Publicado" from "Change to" dropdown
        Then user saves the content page
Scenario: Translate test pages to Spanish to embed Infographic and embed/add Infographic to the test pages
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        Then user selects "Translate" option from Operations dropdown for content with title "Automated Test Article to test Infographic"
        Then page title is "Translations of Home and Landing page for Video"
        When user clicks on "Add" button to add translation
        Then page title is "Crear traducción Español de Automated Test Article to test Infographic"
        And user clears out "Pretty URL" field
        And user fills out the following fields
            | fieldLabel       | value                               | field_name       |
            | Pretty URL       | article-to-test-infographic-spanish | field_pretty_url |
            | Título de página | Spanish                             | title            |
        Then user saves the content page
        And user clicks on the tool bar status button "Borrador"
        And user selects "Quick Publish" from workflow actions
        When user clicks on "Contenido" tab
        Then user selects "Traducir" option from Operations dropdown for content with title "Automated Test Home and Landing page to test Infographic"
        When user clicks on "Add" button to add translation
        Then page title is "Crear traducción Español de Automated Test Home and Landing page to test Infographic"
        And user clears out "Pretty URL" field
        And user fills out the following fields
            | fieldLabel       | value                                             | field_name       |
            | Pretty URL       | home-and-landing-page-to-test-infographic-spanish | field_pretty_url |
            | Título de página | Spanish                                           | title            |
        Then user saves the content page
        And user clicks on the tool bar status button "Borrador"
        And user selects "Quick Publish" from workflow actions
Scenario: Verify Infographic at the front end in the Spanish Article
        Given user is navigating to the front end site with spanish path "/espanol" site section plus "home-and-landing-page-video-spanish"
        And screen breakpoint is set to "desktop"
        And page title is "Automated Test Article to test Infographic Spanish"
        And the intro text reads "Article Intro Text"
        And the caption appears as "Automated Test Infographic Caption"
        And the infographic link "Ver e Imprimir Infografía" appears with href "infographic-desktop"
        And the infographic displayed has the following attributes
            | index | position        | imageSource         | imageAlt | descText                                     | longDesc  |
            | 0     | embedded-entity | infographic-desktop | Alt Text | Infographic Display: Article: Large,None:    | \/espanol |
            | 1     | align-left      | infographic-desktop | Alt Text | Infographic Display: Article: Medium,Left:   | \/espanol |
            | 2     | align-center    | infographic-desktop | Alt Text | Infographic Display: Article: Medium,Center: | \/espanol |
            | 3     | align-right     | infographic-desktop | Alt Text | Infographic Display: Article: Medium,Right:  | \/espanol |
    Scenario: Verify Infographic at the front end in the Spanish Home and Landing page
        Given user is navigating to the front end site with spanish path "/espanol" site section plus "home-and-landing-page-to-test-infographic-spanish"
        And page title is "Automated Test Home and Landing page to test InfographicSpanish"
        And multimedia card row displays an image which matches the earlier selected promo image of Infographic
        And multimedia card row displays the following cards
            | title                              | link                                 |
            | Automated Test Infographic Spanish | {TEST_SITE_SECTION}/test-infographic |
        And list card row title is "List Title"
        And list row displays the following links
            | title                              | link                                 | description                                         |
            | Automated Test Infographic Spanish | {TEST_SITE_SECTION}/test-infographic | Automated Test Infographic List Description Spanish |
        And the thumbnail image has an image which matches the earlier selected promo image of Infographic
    Scenario: Clean up
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user selects a checkbox next to title with url "article-to-test-infographic" from the list of content
        And user selects a checkbox next to title with url "home-and-landing-page-to-test-infographic" from the list of content
        And user clicks on "Apply to selected items" content action button
        Then page title is "Are you sure you want to delete these content items?"
        When user clicks on "Delete" button
        Then the confirmation text "Deleted 4 content items" appears on a screen
        And the content item with url "article-to-test-infographic" does not exist in the list of content
        And the content item with url "home-and-landing-page-to-test-infographic" does not exist in the list of content
        And the content item with url "article-to-test-infographic-spanish" does not exist in the list of content
        And the content item with url "home-and-landing-page-to-test-infographic-spanish" does not exist in the list of content
        And user clicks on "Media" sub tab
        And user selects a checkbox next to title with url "test-infographic" from the list of content
        And user clicks on "Apply to selected items" content action button
        Then page title is "Are you sure you want to delete this media item?"
        When user clicks on "Delete" button
        Then the confirmation text "Deleted 2 items" appears on a screen
        And the content item with url "test-infographic" does not exist in the list of content
        And the content item with url "test-infographic-spanish" does not exist in the list of content
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Closes #429 

Added tests for Infographic translation to $InfographicCreation.feature

### CODE REVIEW

**Comment by alinai** [comment]
Test run on dev-ac:
image

File: cypress/e2e/CMS/$InfographicCreation/index.js [hunk 1]
```diff
@@ -290,4 +295,33 @@ And('the thumbnail image has an image which matches the earlier selected promo i
...
         expect(actSrc).to.include(extractedImageName.replaceAll('_', '-').replace('article', ''))
     });
+});
+
+Then('user selects {string} option from Operations dropdown for media with title {string}', (translateOption, title) => {
+    cy.get(`td:contains('${title}')`).siblings('td').find(`ul.dropbutton >li> a:contains("${translateOption}")`).click({ force: true });
+});
+
+Given('user is navigating to the front end site with spanish path {string} site section plus {string}', (spPath, purl) => {
+    cy.visit(`${frontEndBaseUrl}${spPath}${siteSection}/${purl}-${randomStr}`, { retryOnStatusCodeFailure: true });
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$InfographicCreation/index.js:305]: this exists in common - can be removed

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

File: cypress/e2e/CMS/$InfographicCreation/index.js (hunk 1 above)
Comment (alinai) [reply cypress/e2e/CMS/$InfographicCreation/index.js:305]: done

**Comment by alinai** [comment]
test run on test-ac after review:
image

**Comment by alinai** [comment]
Please review again @KateMashkinaNIH

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**